plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -561,7 +561,8 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
 
             bool disable_subjective_billing = ( _pending_block_mode == pending_block_mode::producing )
                                               || ( persist_until_expired && _disable_subjective_api_billing )
-                                              || ( !persist_until_expired && _disable_subjective_p2p_billing );
+                                              || ( !persist_until_expired && _disable_subjective_p2p_billing )
+                                              || trx->read_only;
```

Comment:
This logic needs to be applied to push_transaction near line 1960. There persist is `itr->trx_type == trx_enum_type::persisted`